TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/NettyPacketToHttpConsumer.java

```diff
@@ -133,6 +133,7 @@ public NettyPacketToHttpConsumer(NioEventLoopGroup eventLoopGroup, URI serverUri
     }
 
     private void addHttpTransactionHandlersToPipeline(ChannelPipeline pipeline) {
+        pipeline.addLast(new BacksideSnifferHandler(responseBuilder));
```

Comment:
I've been trying to understand why BacksideSnifferHandler is a ChannelDuplexHandler when we are only acting on channelRead, would this make it justifiable to be a ChannelInboundHandlerAdapter or is there something else happening here?